src/database-layer/db.ts

```diff
@@ -689,6 +696,24 @@ if (maybePg != null) {
 		return {
 			engine: Engines.postgres,
 			executeSql: atomicExecuteSql,
+			on: async (name, fn, options) => {
+				if (name === 'notification' && options?.channel === undefined) {
+					throw new Error('Missing channel option for notification listener');
+				}
+
+				const client = await pool.connect();
+				client.on(name, async (msg) => {
+					try {
+						await fn(msg);
+					} catch (error) {
+						console.error('Error handling message:', error);
+					}
+				});
+
+				if (name === 'notification' && options?.channel !== undefined) {
+					await client.query(`LISTEN "${options.channel}";`);
```

Comment:
Can this use a bind var?

Reply:
I double-checked, but Postgres doesn't support parameterized queries for `LISTEN`. We could maybe use something like `pg-format` (although its quite old at this point) or Postgres' [format](https://www.postgresql.org/docs/current/functions-string.html#FUNCTIONS-STRING-FORMAT) with `%I` I guess.

Reply:
Will check for double quotes before the `query()` call and throw if any are found in the channel name. Something simple and fast and should be plenty for real use cases.